perfkitbenchmarker/providers/mesos/mesos_docker_instance.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2015 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import requests
+import urlparse
+
+from perfkitbenchmarker import disk
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import virtual_machine, linux_virtual_machine
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.providers.mesos.mesos_disk import LocalDisk
+
+FLAGS = flags.FLAGS
+
+MARATHON_API_PREFIX = '/v2/apps/'
+USERNAME = 'root'
+
+
+class MesosDockerInstance(virtual_machine.BaseVirtualMachine):
+  """
+  Represents a Docker instance spawned by Marathon framework on a Mesos cluster
+  """
+
+  CLOUD = 'Mesos'
+
+  def __init__(self, vm_spec):
+    super(MesosDockerInstance, self).__init__(vm_spec)
+    self.user_name = USERNAME
+    self.api_url = urlparse.urljoin(FLAGS.marathon_address, MARATHON_API_PREFIX)
+    self.app_url = urlparse.urljoin(self.api_url, self.name)
+
+  def _CreateDependencies(self):
+    self._CheckPrerequisites()
+    self._CreateVolumes()
+
+  def _Create(self):
+    self._CreateApp()
+    self._WaitForBootCompletion()
+
+  def _PostCreate(self):
+    self._SetupSSH()
+    self._ConfigureProxy()
+
+  def _Delete(self):
+    self._DeleteApp()
+
+  def _CheckPrerequisites(self):
+    """
+    Exits if any of the prerequisites is not met.
+    """
+    if self.disk_specs and self.disk_specs[0].disk_type == disk.STANDARD:
+      raise Exception('Currently only local disks are supported. Please '
+                      're-run the benchmark with "--scratch_disk_type=local".')
+    if not FLAGS.marathon_address:
+      raise Exception('Please provide the address and port of Marathon '
+                      'framework. Example: 10:20:30:40:8080')
+
+  def _CreateVolumes(self):
+    """
+    Creates volumes for scratch disks.
+    """
+    for disk_num, disk_spec in enumerate(self.disk_specs):
+      if disk_spec.disk_type == disk.LOCAL:
+        scratch_disk = LocalDisk(disk_num, disk_spec, self.name)
+      else:
+        # TODO: support for Ceph
+        pass
```

Comment:
Maybe explicitly raise an exception here that mentions that Ceph isn't supported yet